airbyte_cdk/connector_builder/connector_builder_handler.py

```diff
@@ -90,7 +88,7 @@ def create_source(
 
 
 def read_stream(
-    source: DeclarativeSource,
+    source: ConcurrentDeclarativeSource[Optional[List[AirbyteStateMessage]]],
```

Comment:
out-of-scope for this PR: I always found the generic for `ConcurrentDeclarativeSource` dangerous and since `TState` is always `Optional[List[AirbyteStateMessage]]` and we assume so at a couple of places like [here](https://github.com/airbytehq/airbyte-python-cdk/blob/f525803b3fec9329e4cc8478996a92bf884bfde9/airbyte_cdk/sources/declarative/concurrent_declarative_source.py#L109). I'm wondering why it was put there in the first place and if we should clean that up as well

Reply:
yeah that's a good question. i can't remember why I wrote it this way when we built the class, but a part of me would think I wouldn't introduce generics unless it was absolutely necessary for mypy issues... I'm gonna add a note because I agree unless it's strictly needed, it does more harm than good